packages/dds/sequence/src/intervalCollection.ts

```diff
@@ -562,21 +561,22 @@ export class IntervalCollectionValueType implements IValueType<IntervalCollectio
 }
 
 export function makeOpsMap<T extends ISerializableInterval>(): Map<
-	string,
+	IntervalOpType,
 	IValueOperation<IntervalCollection<T>>
 > {
-	const rebase = (
-		collection: IntervalCollection<T>,
-		op: IValueTypeOperationValue,
-		localOpMetadata: IMapMessageLocalMetadata,
+	const rebase: IValueOperation<IntervalCollection<T>>["rebase"] = (
+		collection,
+		op,
+		localOpMetadata,
 	) => {
 		const { localSeq } = localOpMetadata;
+		assert(op.value !== undefined, "op must have value");
```

Comment:
this is a bit weird, rebaseLocalInterval doesn't accept undefined, but can return. if it is returned then i think we could get an undefined op. no test fail, so i'm not sure its valid. a better fix might be to make rebaseLocalInterval never return undefined, and exclude undefined from the op type, which seems invalid anyway

Reply:
Might be a later discussion, but it looks like rebaseLocalInterval only returns undefined if the interval slid off the string - what would we do in that case instead? I think it makes sense to make sure the interval that the op refers to hasn't slid off the string in this case though.

Reply:
i don't know all the cases where an interval can slide off the string. maybe if the length goes to 0? what happen in that case? do they all slide off? in the reconnect case i guess we just wouldn't send as op, as all the other clients would have removed their intervals as well? i think @connorskees  was looking at something like this as well where we could put intervals that slide off rather than detaching them? can we send an op to directly send on in that position?

Reply:
An interval can slide off the string in two cases determined by a feature flag. By default, endpoint references can become detached only if the entire string is deleted. If you enable the feature flag `mergeTreeReferencesCanSlideToEndpoint`, endpoints will only look in one direction prior to sliding off the string. That is, it doesn't look in the opposite direction if it fails to find something to slide on to. 

You _can_ emit an op to have something go to one of the special endpoint segments by specifying the strings "start" and "end" as the endpoint positions, but right now that wouldn't really accomplish what you're trying to do AFAICT.

Reply:
@connorskees  and @jzaffiro i took this a different way. i now allow the rebase function to return undefined, which means do not send an op, which is allowed on resubmit paths. i think this is better, as i don't think the client will process and op with undefined as the interval value correctly, so better to just not send it.